### PR TITLE
Tic tac toe

### PR DESCRIPTION
**Pull request by lucasmachadorj**
I made this solution based strictly on emergent design.
There are better abstractions but I'll do it in the refactoring exercise with upfront design.

### CODE REVIEW

File: 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts [hunk 1]
```diff
@@ -12,6 +12,10 @@ class Game {
   isBoardEmpty(): boolean {
     return this.board.every((row) => row.every((cell) => cell === ""));
```
Comment (stemmlerjs) [inline 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts]: Niceeee. I like this simplicity.

File: 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts [hunk 2]
```diff
@@ -12,6 +12,10 @@ class Game {
...
   }
+
+  currentTurn(): string {
+    return "X";
```
Comment (stemmlerjs) [inline 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts]: Nice _Fake It_.

File: 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts [hunk 3]
```diff
@@ -28,6 +40,11 @@ describe("Tic tac toe game", () => {
     expect(game.isBoardEmpty()).toEqual(true);
   });
 
+  it("should have a board with 9 cells", () => {
+    const game = new Game();
+    expect(game.boardSize).toEqual(9);
```
Comment (stemmlerjs) [inline 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts:23]: Recommendation: For your commands and queries, don't use 'get' and 'set' (if you really don't have to). Always make them _clear methods_. Force your clients to invoke with ().

Why?

It's not abundantly clear if you need to call a method or if it's a property, from the outside. It's more explicit to do `getBoardSize()`.

It's really easy to burn yourself, thinking you're referring to a property when you should actually be calling a method, and vice-versa. There have been times where I've done this, just referring to a method signature 

I recommend standardizing this, to go from this to:

```typescript
getBoardSize(): number {
    return this.rowsSize * this.columnsSize;
}
```

File: 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts [hunk 4]
```diff
@@ -49,4 +57,10 @@ describe("Tic tac toe game", () => {
     const game = new Game();
     expect(game.currentTurn()).toEqual("X");
   });
+
+  it("should mark the first cell with X when first player chooses the first cell", () => {
+    const game = new Game();
+    game.playAt(0, 0);
+    expect(game.playerAt(0, 0)).toEqual("X");
```
Comment (stemmlerjs) [inline 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts:61]: Nice API decision. I'm assuming you Programming By Wishful Thinking here? AND worked backwards? Seems easy to use.

File: 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+  private isCellEmpty(row: number, column: number): boolean {
+    return this.board[row][column] === "";
+  }
+
+  private get board(): Board {
+    return this.props.board;
+  }
+
+  private verifyWinner(row: number, column: number): void {
```
Comment (stemmlerjs) [inline 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.ts:79]: I love this method. So clean and easily readable. Domain-Driven. Can't misunderstand this.

File: 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts (hunk 3 above)
Comment (lucasmachadorj) [reply 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts:23]: Gotcha, bro
That makes sense

File: 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts (hunk 4 above)
Comment (lucasmachadorj) [reply 1_The_Physical/Level_2_Stateful_Guess_Point/Shared/TicTacToe/src/tictactoe.spec.ts:61]: Yeah, programming by wishful thinking and backwards are helping me a lot to design simple apis

**Pull request merged**